src/history/hs.c

```diff
@@ -894,9 +893,22 @@ __wt_find_hs_upd(
                  *
                  * We keep looking back into history store until we find a base update to apply the
                  * reverse deltas on top of.
+                 *
+                 * If our get key fails here we fall back to the datastore version. If its timestamp
+                 * doesn't match our timestamp then we've failed.
```

Comment:
Is it actually failed? Or is it just that there isn't a match? I guess we've actually failed, since we are applying a modify chain and there is no full value to start from.

Reply:
Not having a match would be a failure in this instance, as far as I can tell? We might unintentionally shadow a more serious failure when the get key fails but we would return error either way.